code-challenges.test.js

```diff
@@ -16,35 +16,140 @@
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
+colors1.shift()
+console.log(colors1)
+// Was able to use .shift to remove the first element in the array.
 
 // b) Create the function that makes the test pass.
 
+// Create a function that takes in an array
+const arrayShuffle = (array) => {
+     let newArr = array.shift()
+// Use a for loop to iterate over the array
+     for (let i = 0; i < array.length; i++) {
+
+ // possibly use math.random to shuffle
+     }
+// return the new array 
+     return newArr.shift
+
+   }
+  
+   console.log(arrayShuffle(colors1))
+   console.log(arrayShuffle(colors1))
+
+// Notes: Note sure how to incorporate math.random into the for loop in order to shuffle. 
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const votes1 = {upVotes: 13, downVotes: 2}
-// Expected output: 11
-const votes2 = {upVotes: 2, downVotes: 33}
+// const votes1 = {upVotes: 13, downVotes: 2,}
+// const votes2 = {upVotes: 2, downVotes: 33}
 // Expected output: -31
 
-
+describe("endTally", () => {
+     it("creates an end talley of two numbers from an object", () => {
+       expect(endTally(votes1)).toEqual(11)
+       expect(endTally(votes2)).toEqual(-31)
+    })
+  })
+    // Good Error - Jest failed
+    // ReferenceError: endTally is not defined
+    // est Suites: 1 failed, 1 total
+    // Tests:       1 failed, 1 total
+
+    // 
+   
+  
+//       ReferenceError: endTally is not defined
+  
+//         61 | describe("endTally", () => {
+//         62 |     it("creates an end talley of two numbers from an object", () => {
+//       > 63 |       expect(endTally(votes1)).toEqual(11)
+//            |       ^
+//         64 |       expect(endTally(votes2)).toEqual(-31)
+//         65 |    })
+//         66 |  })
+  
+//         at Object.expect (code-challenges.test.js:63:7)
+  
+//   Test Suites: 1 failed, 1 total
+//   Tests:       1 failed, 1 total
 // b) Create the function that makes the test pass.
 
 
+const votes1 = {upVotes: 13, downVotes: 2,
+    // Expected output: 11
+    endTally: function() {
+        return this.upVotes - this.downVotes
+    // create a function within const votes using .this and the values of upVotes and downVotes
+    // return subtraction operator .upvotes minus .downvotes to receive the tally
+    }}
+const votes2 = {upVotes: 2, downVotes: 33,
+        // Expected output: -31
+
+    endTally: function() {
+        return this.upVotes - this.downVotes
+    // create a function within const votes using .this and the values of upVotes and downVotes
+    // return subtraction operator .upvotes minus .downvotes to receive the tally
+        }}
+
+    console.log(votes1.endTally())
+    console.log(votes2.endTally())
+
+// Notes: Console log returns correct ouput
```

Comment:
What is your test showing?  For this problem, we want to create a single function that takes the object as an argument.